usaspending_api/references/models.py

```diff
@@ -401,10 +470,303 @@ class LegalEntity(DataSourceTrackedModel):
     individual = models.TextField(blank=True, null=True)
 
     def save(self, *args, **kwargs):
+        LegalEntity.update_business_type_categories(self)
         super(LegalEntity, self).save(*args, **kwargs)
 
         LegalEntityOfficers.objects.get_or_create(legal_entity=self)
 
+    @staticmethod
+    def update_business_type_categories(le):
+        # Business Category
+        le.category_small_business = (
+            le.small_business == "1" or
+
+            le.business_types == "R"  # For-Profit Organization (Other than Small Business)
+        )
+        le.category_other_than_small_business = (
+            le.business_types == "Q"  # For-Profit Organization (Other than Small Business)
+        )
+        le.category_business = (
+            le.for_profit_organization == "1" or
+
+            le.category_small_business or
+            le.category_other_than_small_business
+        )
+        # End Business Category
+
+        # Minority Owned Business Category
+        le.category_alaskan_native_owned_business = (
+            le.alaskan_native_owned_corporation_or_firm == "1"
```

Comment:
Very broadly, on all these fields, have we verified that we don't have any "uncleansed" versions of the same data?  For instance, if we're looking for "1", might there be variants like "1 ", "1.0", or maybe "Y", "T", etc., hiding out in our data?  And, for the ones holding letters, that there aren't lowercase variants?